src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -157,42 +188,52 @@ void KeyboardManagerState::UpdateDetectSingleKeyRemapUI()
     }
 
     std::unique_lock<std::mutex> detectedRemapKey_lock(detectedRemapKey_mutex);
-    hstring remapKeyString = winrt::to_hstring((unsigned int)detectedRemapKey);
+    hstring key = winrt::to_hstring((unsigned int)detectedRemapKey);
     detectedRemapKey_lock.unlock();
 
     // Since this function is invoked from the back-end thread, in order to update the UI the dispatcher must be used.
-    currentSingleKeyRemapTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [=]() {
-        currentSingleKeyRemapTextBlock.Text(remapKeyString);
+    currentSingleKeyRemapTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [this, key]() {
+        currentSingleKeyRemapTextBlock.Children().Clear();
+        AddKeyToLayout(currentSingleKeyRemapTextBlock, key);
+        currentSingleKeyRemapTextBlock.UpdateLayout();
     });
 }
 
 // Function to return the currently detected shortcut which is displayed on the UI
 std::vector<DWORD> KeyboardManagerState::GetDetectedShortcut()
 {
     std::unique_lock<std::mutex> lock(currentShortcutTextBlock_mutex);
-    hstring detectedShortcutString = currentShortcutTextBlock.Text();
+
+    std::vector<DWORD> keys;
+    if (currentShortcutTextBlock.Children().Size() > 0)
+    {
+        for (auto border : currentShortcutTextBlock.Children())
+        {
+            auto keyString = border.as<Border>().Child().as<TextBlock>().Text();
+            keys.push_back(std::stoul(keyString.c_str()));
+        }    
+    }
+

```

Comment:
Some minor changes based on the latest changes from the dev branch will be needed here as well.